input-output-hk/lace
# feat(staking): [LW-8751] show pool rewards in activity

# Checklist

- [x] JIRA - LW-8751

---

## Proposed solution

Join rewardsHistory with wallet activity observable data and show it in wallet activity. The logic around showing the activity was at this point pretty complex and heavily relied on the activity being bound to a transaction which isn't the case for pool rewards which are just deposited to user's reward account at the end of the epoch. To avoid confusion down the round around that nuance, this PR also refactors the `TransactionDetailBrowser` component and the related "sub-components"/state in way that rewards activity is clearly split from "transactions" activity in a type-safe way

The overall diff is pretty big and I would recommend reviewing this PR primarily by commits - the first commit introduces the rewards in the history in the least "invasive" way and the subsequent commits refactor the components/state in several stages, achieving the desired split of rewards from transactions activity at the end.

## Testing

* Check that rewards are properly shown and check that the timestamps correspond with respective epoch starts, ideally by comparing with a block explorer
* Check that wallet activity is properly loaded, not only for rewards but also for other kinds of transactions which this PR touched but there should be no functional change.

Note on designs: there are no Figma designs to validate the "Rewards" detail screen against. As per the conversation in the JIRA task, the Rewards detail screen has been implemented reusing existing elements as much as possible. Feel free to suggest improvements to the looks/layout of the Rewards detail screen

## Screenshots

![Screenshot 2023-10-17 at 00 08 43](https://github.com/input-output-hk/lace/assets/4980147/16324910-0d38-4eb8-8069-9185ab679316)

![Screenshot 2023-10-17 at 00 08 49](https://github.com/input-output-hk/lace/assets/4980147/084d9f79-33a7-4597-ac9c-7d5e730c5489)

in case of rewards from multiple pools:
![Screenshot 2023-10-17 at 00 12 24](https://github.com/input-output-hk/lace/assets/4980147/632fe92e-7749-4f7e-b844-91a5d9c2c451)





---
# Allure report
`allure-report-publisher` generated test report!



**smokeTests**: ❌ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/2561/6536300953/index.html) for [8d20a7b9](https://github.com/input-output-hk/lace/pull/624/commits/8d20a7b9835a49362eb5ef2ea2044408e33b4d97)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 31     | 1      | 0       | 0     | 32    | ❌     |
